delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resources.scala

```diff
@@ -364,4 +366,24 @@ object Resources {
       case c: DeprecateResource => deprecate(c)
     }
   }
+
+  /**
+    * Create an instance of [[EventExchange]] for [[ResourceEvent]].
+    * @param resources  resources operation bundle
+    */
+  def eventExchange(resources: Resources): EventExchange[ResourceEvent] = new EventExchange[ResourceEvent] {
+    override protected def fetchExpanded(event: ResourceEvent): Task[ResourceF[ExpandedJsonLd]] =
+      resources
+        .fetch(IriSegment(event.id), event.project, None)
+        .map(res => res.map(_.expanded))
+        .hideErrorsWith(rej => new IllegalArgumentException(rej.reason))
```

Comment:
I'm not sure about converting all errors to `IllegalArgumentException`, but I don't have a quick solution in mind